tests/integration/kubernetes/confidential/confidential_common.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bats
+# Copyright 2022-2023 Advanced Micro Devices, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+source "../../tests_common.bash"
+
+setup_unencrypted_confidential_pod() {
+    local pod_name="unencrypted-confidential"
+    local IMAGE_REPO="ghcr.io/confidential-containers/test-container"
+    local UNENCRYPTED_IMAGE_URL="${IMAGE_REPO}:unencrypted"
+    local TEST_DIR="$(mktemp -d /tmp/test-kata-confidential.XXXXXXXX)"
+    export SSH_KEY_FILE="${TEST_DIR}/container-ssh-key"
+
+   	k8s_generate_service_yaml "${pod_config_dir}/${pod_name}.yaml" "${UNENCRYPTED_IMAGE_URL}"
+
+    if [ "${KATA_HYPERVISOR}" == "qemu-sev" ] || [ "${KATA_HYPERVISOR}" == "qemu-snp" ]; then
+        # Pull image
+        docker pull "${UNENCRYPTED_IMAGE_URL}"
+        
+        # Get ssh key from docker image label and save to file
+        docker inspect ${image_url} \
+            | jq -r '.[0].Config.Labels.ssh_key' \
+            | sed "s|\(-----BEGIN OPENSSH PRIVATE KEY-----\)|\1\n|g" \
+            | sed "s|\(-----END OPENSSH PRIVATE KEY-----\)|\n\1|g" \
+            > "${SSH_KEY_FILE}"
+
+        # Set permissions on private key file
+        chmod 600 "${SSH_KEY_FILE}"
+
+        if [ "${KATA_HYPERVISOR}" == "qemu-sev" ]; then
+            export DMESG_STRING="SEV"
+        elif [ "${KATA_HYPERVISOR}" == "qemu-snp" ]; then
+            export DMESG_STRING="SNP"
+        fi
+    fi
```

Comment:
On CCv0, I cannot see any use for this block for SE. The test `pull_unencrypted_image_and_set_keys` is not part of the SE tests. @stevenhorsman Are you planning something to add this kind of test for SE?

Reply:
@BbolroC Could we do something like:
`cat /sys/firmware/uv/prot_virt_guest` and check the value is `1` and/or `cat /proc/cpuinfo | grep 158` to check we are in an SE guest?

Reply:
I am thinking instead of doing dmesg | grep $string. I can call a function verify_secure_enclave.
And in `confidential_common.sh` we can define how to verify each env based on `${KATA_HYPERVISOR}`

Reply:
In some other comment here I suggested creating some sort of `execute_in_guest` function that would allow you to run anything via SSH. That could come in handy when we need to use different commands to verify different platforms.

Reply:
For SE, it would be 

```
# cat /sys/firmware/uv/prot_virt_guest
1
```

Reply:
I will change this code where this part will just export the remote command to be executed and then in the bats file, I will just run the remote command inside the confidential container.

Reply:
That's a good approach, and I think all the TEEs can do the check using `cat /something/something`.

I think we can even have the cmds to run declared as
qemu-sev_cmd
qemu-snp_cmd
qemu-tdx_cmd
clh-tdx_cmd
qemu-se_cmd

For now, of course, just add / create the qemu-sev_cmd / qemu-snp_cmd

By doing this we can just call ${${KATA_HYPERVISOR}_cmd} and be happy, I think.

Reply:
Since SEV and SNP requires grep-ing the dmesg buffer and other TEEs might need to cat a file. I have introduced a variable `REMOTE_COMMAND` which will be executed inside the container when running. Thus, when adding new TEE, one just needs to add a command that verifies the execution in TEE. Second commit, which adds SNP testing is an example.